### PR TITLE
feat:Add custom Markdown components to better display Markdown elements, especially code blocks.

### PR DESCRIPTION
**Pull request by Onelevenvy**
#### 1.Dependency requirements: 
`pnpm install react-markdown rehype-highlight highlight.js`
react-markdown is used for parsing Markdown syntax, while rehype-highlight and highlight.js are used for highlighting code blocks.
#### 3.Add a CopyButton for copying code blocks.

Dark mode
image

Light mode
image

### CODE REVIEW

File: frontend/src/components/Markdown/Markdown.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                  align="center"
+                  justify="space-between"
+                  bg={secBgColor}
+                  p={1}
+                  borderBottomWidth="1px"
+                >
+                  <Flex align="center" gap={2}>
+                    <FiTerminal size={18} />
+                    <Text fontSize="sm" color={textColor}>
+                      {node?.data?.meta}
```
Comment (StreetLamb) [inline frontend/src/components/Markdown/Markdown.tsx:37]: Getting error: `Property 'meta' does not exist on type 'ElementData'.`
Maybe we can replace it with the language?
Example:
```js
const match = (className || "").match(/language-(\w+)/);
const language = match ? match[1] : null;
...
<Text fontSize="sm" color={textColor}>
{language}
</Text>
```

File: frontend/src/components/Markdown/Markdown.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import ReactMarkdown from "react-markdown";
+import rehypeHighlight from "rehype-highlight";
+import "highlight.js/styles/atom-one-dark.css";
```
Comment (StreetLamb) [inline frontend/src/components/Markdown/Markdown.tsx:3]: Is it possible to make the theme dynamic depending on whether it's light or dark mode? Currently, in light mode the code is not very readable.

Comment (Onelevenvy) [reply frontend/src/components/Markdown/Markdown.tsx]: Hello,Thank you for your reply. The issue should be resolved now, and I would appreciate it if you could review the code. As a beginner in programming, I'm open to any feedback on inconsistencies or areas for improvement. Please feel free to offer any advice or suggestions.

Comment (Onelevenvy) [reply frontend/src/components/Markdown/Markdown.tsx]: image
It is better now

File: frontend/src/components/Markdown/Markdown.tsx (hunk 1 above)
Comment (Onelevenvy) [reply frontend/src/components/Markdown/Markdown.tsx:54]: Sorry for the inconvenience. I need more information, as I haven't encountered this error

Comment (StreetLamb) [reply frontend/src/components/Markdown/Markdown.tsx:54]: No worries! The build is failing as typescript is complaining that `node.data`'s `ElementData` type does not have a property called `meta`. You can check out the error logs in the last [test job](https://github.com/StreetLamb/tribe/actions/runs/10535248716/job/29197100710).

Comment (Onelevenvy) [reply frontend/src/components/Markdown/Markdown.tsx:54]: Thans，recently discovered this issue and have updated it to display the name of the programming language instead. it is WIP
image

File: frontend/src/components/Markdown/Markdown.tsx (hunk 2 above)
Comment (StreetLamb) [reply frontend/src/components/Markdown/Markdown.tsx]: Thanks! Just for consistency - Could you use `"highlight.js/styles/github.css"` for light mode so it is similar to dark mode's theme?

File: frontend/src/components/Markdown/Markdown.tsx (hunk 1 above)
Comment (Onelevenvy) [reply frontend/src/components/Markdown/Markdown.tsx]: Thank you for the great suggestion; I've made the changes and updated the theme to use 'highlight.js/styles/github.css' for light mode to match the dark mode theme

File: frontend/src/components/Markdown/Markdown.tsx (hunk 2 above)
Comment (Onelevenvy) [reply frontend/src/components/Markdown/Markdown.tsx]: OK,thank you for the great suggestion; I've made the changes.

**Review by StreetLamb — APPROVED** [review]
Looks great! Thanks @Onelevenvy

**Pull request merged**